lib/diagnostic.js

```diff
@@ -5,17 +5,22 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current ) {
+let i =0;
+let minimum = previous.reduce(i++,previous,[0],current);
```

Comment:
```previous``` and ```current``` are assumed to be numbers and trying to call an array method, ```reduce```, on them will throw an error.